### PR TITLE
Multi CPU Arch - add IBM Power and Z support

### PR DESCRIPTION
**Pull request by michal-at-travisci**
Added IBM Power and Z relevant entries and changes
Added identification section
Extended Security & LXD section
Added huge pages support remark

### CODE REVIEW

File: user/multi-cpu-architectures.md [hunk 1]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+> This is an alpha stage of this feature and we are eager to hear back from you, both for [`Arm`-based feedback](https://travis-ci.community/t/about-the-arm-cpu-architecture-category/5336) and [`IBM`-based feedback](link TBD). The definition keys used in the `.travis.yml` file may be further adapted on short notice.
 {: .alpha}
 
-> `Arm`-based building is only available for Open Source repositories (at both travis-ci.org and travis-ci.com). While available to all Open Source repositories, the concurrency available for `Arm`-based jobs is limited during the alpha period.
-> An attempt to run `Arm`-based build for a private repository will result in a build run on standard, non-`Arm` infrastructure. For any commercial queries with regards to multi-arch builds before they are available, please [contact us](mailto:support@travis-ci.com).
+> `IBM Power`, `IBM Z` and `Arm`-based building is only available for Open Source repositories (at both travis-ci.org and travis-ci.com). While available to all Open Source repositories, the concurrency available for multi cpu arch-based jobs is limited during the alpha period.
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
> `IBM Power`, `IBM Z` and `Arm`-based building is only available for Open Source repositories (at both travis-ci.org and travis-ci.com). While available to all Open Source repositories, the concurrency available for multiple CPU arch-based jobs is limited during the alpha period.
```

File: user/multi-cpu-architectures.md [hunk 2]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+> An attempt to run `IBM Power`, `IBM Z` and `Arm`-based build for a private repository will result in a build run on standard, `AMD`-based infrastructure. For any commercial queries with regards to multi-arch builds before they are available, please [contact us](mailto:support@travis-ci.com).
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
> An attempt to run `IBM Power`, `IBM Z` and `Arm`-based builds for a private repository will result in a build run on standard, `AMD`-based infrastructure. For any commercial queries with regards to multi-arch builds before they are available, please [contact us](mailto:support@travis-ci.com).
```

File: user/multi-cpu-architectures.md [hunk 3]
```diff
@@ -4,40 +4,61 @@ layout: en
...
 
-If your code is used on multiple CPU architectures it probably should be tested on multiple CPU architectures. Travis CI can test on amd64 and arm64 (run on ARMv8 compliant CPUs) if the operating system is Linux.
+If your code is used on multiple CPU architectures it probably should be tested on multiple CPU architectures. Travis CI can test on amd64, ppc64le (IBM Power CPUs), s390x (IBM Z CPUs) and arm64 (run on ARMv8 compliant CPUs) if the operating system is Linux.
+
+## Default CPU architecture
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
## Default CPU Architecture
```

File: user/multi-cpu-architectures.md [hunk 4]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+
+The default CPU architecture used in Travis CI builds is `amd64`. It is used when no `arch` key is present. 
+
+## Identifying CPU architecture of build jobs
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
## Identifying CPU Architecture of Build Jobs
```

File: user/multi-cpu-architectures.md [hunk 5]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+
+You can identify for which CPU architecture a build job is run via
+
+- GUI
+  - in build job list there's a specific label and architecture name based on `arch` tag value
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
  - in the build job list, there's a specific label and architecture name based on `arch` tag value
```

File: user/multi-cpu-architectures.md [hunk 6]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+  - in build job view the same specific label is displayed near operating system identifier
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
  - in the build job view, the same specific label is displayed near the operating system identifier
```

File: user/multi-cpu-architectures.md [hunk 7]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+- A default environmental variable printed out during your build job: `$TRAVIS_CPU_ARCH` (for complete list of available default environmental variables please see [Environment Variables - Default Environment Variables](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) documentation).
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
- A default environmental variable printed out during your build job: `$TRAVIS_CPU_ARCH` (for a complete list of available default environmental variables please see our [Environment Variables - Default Environment Variables](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) documentation. 
```

File: user/multi-cpu-architectures.md [hunk 8]
```diff
@@ -4,40 +4,61 @@ layout: en
...
+
+You can identify for which CPU architecture a build job is run via
+
+- GUI
+  - in build job list there's a specific label and architecture name based on `arch` tag value
+  - in build job view the same specific label is displayed near operating system identifier
+- A default environmental variable printed out during your build job: `$TRAVIS_CPU_ARCH` (for complete list of available default environmental variables please see [Environment Variables - Default Environment Variables](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) documentation). 
+
+
+## Enable testing on multiple CPU architectures
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
## Testing on Multiple CPU Architectures
```

File: user/multi-cpu-architectures.md [hunk 9]
```diff
@@ -4,40 +4,61 @@ layout: en
...
 ```
 {: data-file=".travis.yml"}
 
 If you are already using a [build matrix](/user/customizing-the-build/#build-matrix) to test multiple versions, the `arch` key also multiplies the matrix.
 
-- The arm64 CPU architecture build job is run in an LXD compliant Linux OS image. The default image supported by Travis CI is Ubuntu Bionic 18.04.
-
+- The ppc64le (IBM Power) and s390x (IBM Z) build jobs are run in an LXD compliant Linux OS image. 
+- The arm64 CPU architecture build job is run in an LXD compliant Linux OS image.
+- The default LXD image supported by Travis CI is Ubuntu Xenial 16.04 and by using `dist` one can select different supported LXD image. See [CI Environment Overview - Virtualisation Envionment vs Operating System](https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system) doc.
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
- The default LXD image supported by Travis CI is Ubuntu Xenial 16.04 and by using `dist` you can select different supported LXD images. Also see our [CI Environment Overview - Virtualisation Environment vs Operating System](https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system) documentation.
```

File: user/multi-cpu-architectures.md [hunk 10]
```diff
@@ -82,9 +117,9 @@ jobs:
 
 For example, the above `.travis.yml`, would result in running both jobs with the environmental variable LIB_PATH assigned different values being run only on `amd64` architecture.
 
-## Using Docker in `Arm`-Based Builds within LXD Containers
+## Using Docker in Multi CPU Architecture-Based Builds within LXD Containers
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
## Using Docker in Multiple CPU Architecture-Based Builds within LXD Containers
```

File: user/multi-cpu-architectures.md [hunk 11]
```diff
@@ -65,12 +86,26 @@ jobs:
 ```
 {: data-file=".travis.yml"}
 
+Similarly, this matrix would route builds to the ppc64le (IBM Power) and s390x (IBM Z) architecture environments:
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
Similarly, this matrix would route builds to the `ppc64le` (IBM Power) and `s390x` (IBM Z) architecture environments:
```

File: user/multi-cpu-architectures.md [hunk 12]
```diff
@@ -82,9 +117,9 @@ jobs:
 
 For example, the above `.travis.yml`, would result in running both jobs with the environmental variable LIB_PATH assigned different values being run only on `amd64` architecture.
 
-## Using Docker in `Arm`-Based Builds within LXD Containers
+## Using Docker in Multi CPU Architecture-Based Builds within LXD Containers
 
-It is possible to use Docker in `Arm`-based builds within an LXD container. You may need an arm64v8 docker image as a base or ensure arm64 libraries required by your build are added to your Dockerfile. In order to use docker in `Arm`-based build within an LXD container, docker commands must be run with `sudo`.
+It is possible to use Docker in Multi CPU Architecture-based builds within an LXD container. You may need an specific CPU architecture compliant docker image as a base or ensure relevant libraries required by your build are added to your Dockerfile.
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
It is possible to use Docker in multiple CPU architecture-based builds within an LXD container. You may need a specific CPU architecture compliant docker image as a base or ensure relevant libraries required by your build are added to your Dockerfile.
```

File: user/multi-cpu-architectures.md [hunk 13]
```diff
@@ -108,10 +143,14 @@ script: docker run my/test #assuming docker image my/test is arm64v8 ready
 ```
 {: data-file=".travis.yml"}
 
+You can try it out also for `ppc64le` (IBM Power) and `s390x` (IBM Z) based docker builds, assuming all dependencies and/or CPU architecture compliant base docker image is used.
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
You can try it out also for `ppc64le` (IBM Power) and `s390x` (IBM Z) based docker builds, assuming all dependencies and/or a CPU architecture compliant base docker image are used.
```

File: user/multi-cpu-architectures.md [hunk 14]
```diff
@@ -108,10 +143,14 @@ script: docker run my/test #assuming docker image my/test is arm64v8 ready
...
+
 You can also have a look at [Using Docker in Builds](user/docker/).
 
 ## Security and LXD Container
 
+### Access to priviliged fs/features (Apparmor)
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
### Access to Privileged fs/Features (Apparmor)
```

File: user/multi-cpu-architectures.md [hunk 15]
```diff
@@ -123,3 +162,16 @@ sudo docker run --privileged --rm -t -v /sys/kernel/debug:/sys/kernel/debug:rw
 would result in an error.
 
 Also have a look at the [Github issue relevant to the topic](https://github.com/lxc/lxd/issues/2661) and the [LXD apparmor setup](https://github.com/lxc/lxd/blob/master/lxd/apparmor/apparmor.go) for more details.
+
+### System calls interception
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
### System Calls Interception
```

File: user/multi-cpu-architectures.md [hunk 16]
```diff
@@ -123,3 +162,16 @@ sudo docker run --privileged --rm -t -v /sys/kernel/debug:/sys/kernel/debug:rw
...
+
+
+If you run into message like:
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
If you run into a message like:
```

File: user/multi-cpu-architectures.md [hunk 17]
```diff
@@ -123,3 +162,16 @@ sudo docker run --privileged --rm -t -v /sys/kernel/debug:/sys/kernel/debug:rw
...
+
+> System doesn't support syscall interception
+
+It most probably means a system call interception is outside of the list of the ones considered to be safe (LXD can allow system call interception [if it's considered to be safe](https://github.com/lxc/lxd/blob/master/doc/syscall-interception.md)). 
+
+## Hugepages support from within LXD Container
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
## Hugepages Support from within LXD Container
```

File: user/multi-cpu-architectures.md [hunk 18]
```diff
@@ -123,3 +162,16 @@ sudo docker run --privileged --rm -t -v /sys/kernel/debug:/sys/kernel/debug:rw
...
+
+Build job can’t enable hugepages within LXD container - this is something that may change in the future, yet it depends on potential Linux kernel changes, which is something that needs to be reviewed and developed.
```
Comment (Pezi777) [inline user/multi-cpu-architectures.md]:
```suggestion
A build job can’t enable hugepages within an LXD container - this is something that may change in the future, yet it depends on potential Linux kernel changes, which is something that needs to be reviewed and developed.
```

File: user/multi-cpu-architectures.md (hunk 9 above)
Comment (michal-at-travisci) [reply user/multi-cpu-architectures.md]: Thanks! Added that one manually

**Pull request merged**